neo/public/javascripts/cache/services.js

```diff
@@ -0,0 +1,356 @@
+ angular.module('cloudberry.cache', ['leaflet-directive', 'cloudberry.common' ])
+ .service('Cache', function( $window, $http, $compile, Asterix){
+
+
+  var cachedCityPolygonTree = rbush();
+  var cachedRegion ;
+  var cacheSize = 0;
+  var insertedTreeIDs = new Set();
+  var cacheThreshold = 5500;//soft limit
+  var DeleteTarget = 0;
+  var DeletedCount = 0;
+  var preFetchDistance = 25;
+
+
+
+  this.getCityPolygonsFromCache = function city(bounds){
```

Comment:
Put a comment about this function. What does this function do? This applies to the all functions.